TreeMaker/python/makeTreeFromMiniAOD_cff.py

```diff
@@ -97,9 +97,12 @@ def makeTreeFromMiniAOD(self,process):
     ## PDF weights for PDF systematics
     ## ----------------------------------------------------------------------------------------------
     if self.geninfo and self.doPDFs:
-        process.PDFWeights = cms.EDProducer('PDFWeightProducer')
-        self.VectorDouble.extend(['PDFWeights:PDFweights','PDFWeights:ScaleWeights'])
-        self.VectorInt.extend(['PDFWeights:PDFids'])
+        from TreeMaker.Utils.pdfweightproducer_cfi import PDFWeightProducer
+        process.PDFWeights = PDFWeightProducer.clone(
+            recalculatePDFs = cms.bool(self.signal),
+            pdfSetName = cms.string("NNPDF31_lo_as_0130"),
```

Comment:
Do you know how often this name will change? Is there any way to get it from the tune name or the sample automatically?

Reply:
Unfortunately not, as far as I can tell. See my long hypernews post for more details. The only way to get this info is parsing the massive LHE header block, and that seems not entirely reliable to me (in addition to being overly complicated).

Reply:
This is the one used in TuneCP2.